apps/prairielearn/src/lib/editors.ts

```diff
@@ -1162,28 +1162,45 @@ export class QuestionModifyEditor extends Editor {
 }
 
 export class QuestionDeleteEditor extends Editor {
-  private question: Question;
+  private questions: Question[];
 
-  constructor(params: BaseEditorOptions) {
+  constructor(params: BaseEditorOptions & { questions: Question | Question[] }) {
     super(params);
 
-    this.question = params.locals.question;
-    this.description = `Delete question ${this.question.qid}`;
+    if (Array.isArray(params.questions)) {
+      this.questions = params.questions;
+    } else {
+      this.questions = [params.questions];
+    }
+
+    this.description =
+      this.questions.length === 1
+        ? `Delete question ${this.questions[0].qid}`
+        : `Delete questions ${this.questions.map((x) => x.qid).join(', ')}`;
   }
 
   async write() {
-    assert(this.question.qid, 'question.qid is required');
-
     debug('QuestionDeleteEditor: write()');
-    await fs.remove(path.join(this.course.path, 'questions', this.question.qid));
-    await this.removeEmptyPrecedingSubfolders(
-      path.join(this.course.path, 'questions'),
-      this.question.qid,
-    );
+
+    const qids: string[] = [];
+
+    for (const question of this.questions) {
+      assert(question.qid, 'question.qid is required');
+      //never will be false, but this is to placate typechecks
+      await fs.remove(path.join(this.course.path, 'questions', question.qid));
+      await this.removeEmptyPrecedingSubfolders(
+        path.join(this.course.path, 'questions'),
+        question.qid,
+      );
+      qids.push(question.qid);
+    }
 
     return {
-      pathsToAdd: [path.join(this.course.path, 'questions', this.question.qid)],
-      commitMessage: `delete question ${this.question.qid}`,
+      pathsToAdd: qids.map((qid) => path.join(this.course.path, 'questions', qid)),
```

Comment:
Nit: if you map over `questions` instead, you can save a few lines and some redundant computations and get rid of the `qids` variable entirely.

Reply:
Done in 16aa671758e5d243976b2f58594e0032923eb6f8.